spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/exceptions/OfflineWorkingTimeExceedingException.java

```diff
@@ -0,0 +1,35 @@
+package eu.coatrack.proxy.security.exceptions;
+
+/*-
+ * #%L
+ * coatrack-proxy
+ * %%
+ * Copyright (C) 2013 - 2021 Corizon | Institut für angewandte Systemtechnik Bremen GmbH (ATB)
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import org.springframework.security.core.AuthenticationException;
+
+/**
+ * Indicates that the maximum time the gateway is allowed to work in offline mode
+ * was exceeded. This means that locally cached API keys should be considered
+ * invalid until the online mode is re-established.
+ */
+public class OfflineWorkingTimeExceedingException extends RuntimeException {
```

Comment:
Should extend `AuthenticationException`

Reply:
This case does not say anything about an API keys validity, it only makes them being rejected. Therefore the convention "validity can not be decided" should be applied indirectly via RuntimeException.